metadata/src/main/java/org/apache/kafka/image/ImageProvenance.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.image;
+
+import java.util.Objects;
+
+
+/**
+ * The source of a metadata image.
+ */
+public final class ImageProvenance {
+    public static final ImageProvenance EMPTY = new ImageProvenance("the empty image",
+            0L,
+            0,
+            0L);
+
+    private final String source;
+    private final long highestOffset;
+    private final int highestEpoch;
+    private final long highestTimestamp;
+
+    public ImageProvenance(
+        String source,
+        long highestOffset,
+        int highestEpoch,
+        long highestTimestamp
```

Comment:
Can we document all of these fields? I specifically don't understand how to use the `source` field.